tests/unit/misc/test_cmdhistory.py

```diff
@@ -0,0 +1,196 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2015 Alexander Cogneau (acogneau) <alexander.cogneau@gmail.com>:
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests for misc.History."""
+
+import pytest
+
+from qutebrowser.misc.cmdhistory import (History, HistoryEmptyError,
+                                         HistoryEndReachedError)
+
+
+HISTORY = ['first', 'second', 'third', 'fourth', 'fifth']
+
+CONFIG_NOT_PRIVATE = {'general': {'private-browsing': False}}
+CONFIG_PRIVATE = {'general': {'private-browsing': True}}
+
+
```

Comment:
I don't think using classes is a good idea here - I usually use them, if there are many "groups" of different tests (with different fixtures, setup/teardown, etc.), or if there's a clear mapping of "one test class with many tests per tested function".

Here it would be easier to simply write test functions without any classes (and drop the `self` of course).

To get the history objects, you can use pytest fixtures which return the history:

```python
@pytest.fixture
def hist():
    return cmdhistory.History(history=HISTORY)


def test_history(hist):
    assert hist.history == HISTORY


def test_is_browsing(hist):
    hist._tmphist = None
    assert not hist.is_browsing()

    hist._tmphist = HISTORY
    assert hist.is_browsing()

# ...
```